dependencies/osx/install-dependencies-osx-arch

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+#
+# install-dependencies-osx-arch
+#
+# Copyright (C) 2021 by RStudio, PBC
+#
+# Unless you have received this program directly from RStudio pursuant
+# to the terms of a commercial license agreement with RStudio, then
+# this program is licensed to you under the terms of version 3 of the
+# GNU Affero General Public License. This program is distributed WITHOUT
+# ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+# MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+# AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+#
+#
+
+set -eux
+
+# NOTE: the intention is for this script to be invoked as e.g.
+#
+#   arch -x86_64 ./install-dependencies-osx-arch
+#   arch -arm64  ./install-dependencies-osx-arch
```

Comment:
Is there a way to detect and/or fail the script quickly if it wasn't invoked correctly?

Reply:
Not really -- if not run through `arch`, then the install script just runs for whichever architecture is currently "active". This is what you want on x86_64 (where there's only one possible architecture); on arm64 this is useful when you want to ensure a particular script is run either natively or with the Rosetta translation layer active.